### PR TITLE
Fixed issue with default url

### PR DESCRIPTION
**Pull request by mpminayo**
Changed config file so that it can use default authority when none is provided.

### CODE REVIEW

**Comment by jasonnutter** [comment]
Is the issue here when empty string or null is provided as the authority?

**Comment by jasonnutter** [comment]
Also, can you please add a test that demonstrates this is fixed?

File: lib/msal-node/src/config/Configuration.ts [hunk 1]
```diff
@@ -129,7 +129,12 @@ export function buildAppConfiguration({
     system,
 }: Configuration): NodeConfiguration {
     return {
-        auth: { ...DEFAULT_AUTH_OPTIONS, ...auth },
+        auth: { ...DEFAULT_AUTH_OPTIONS, 
+            ...auth , 
+            authority: auth.authority?.trim() 
+                ? auth.authority.trim() 
+                : DEFAULT_AUTH_OPTIONS.authority
```
Comment (jasonnutter) [inline lib/msal-node/src/config/Configuration.ts]: Ternary not needed if the condition is also the value:

```suggestion
            authority: auth.authority?.trim() || DEFAULT_AUTH_OPTIONS.authority
```

**Comment by mpminayo** [comment]
> Is the issue here when empty string or null is provided as the authority?

Yes it is.

**Comment by mpminayo** [comment]
> Also, can you please add a test that demonstrates this is fixed?

On it.

**Review by samuelkubai — APPROVED** [review]
LGTM

File: lib/msal-node/test/config/ClientConfiguration.spec.ts [hunk 2]
```diff
@@ -69,6 +69,17 @@ describe('ClientConfiguration tests', () => {
         expect(config.auth!.clientId).toEqual(TEST_CONSTANTS.CLIENT_ID);
     });
 
+    test('builds configuration and assigns default authority when empty string provided', () => {
+        const config: Configuration = buildAppConfiguration({
+            auth: {
+                clientId: TEST_CONSTANTS.CLIENT_ID,
+                authority: " "
```
Comment (sameerag) [inline lib/msal-node/test/config/ClientConfiguration.spec.ts:76]: Do we have a test case for `authority=""`?

**Review by sameerag — APPROVED** [review]
Approved with comments.

File: lib/msal-node/test/config/ClientConfiguration.spec.ts (hunk 2 above)
Comment (mpminayo) [reply lib/msal-node/test/config/ClientConfiguration.spec.ts:76]: Just added this.

Comment (jasonnutter) [reply lib/msal-node/test/config/ClientConfiguration.spec.ts:76]: In common, we have this logic for empty string: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/8c7c0ea26f4f42078e2ef62cebe9a3f6c88dd261/lib/msal-common/src/authority/AuthorityFactory.ts#L50-L52

It seems like today, empty string is treated as invalid use case, as we throw an error. Seems like we should be consistent between all of our libraries about this?

Comment (jasonnutter) [reply lib/msal-node/test/config/ClientConfiguration.spec.ts:76]: In other words, it seems like the change to fallback to the default authority should be done in msal-common, as opposed to in MSAL Node?

**Comment by ghost** [comment]
Reminder: The next release is scheduled for next week and this PR appears to be stale :( 

If changes have been requested please address feedback. 
If this PR is still a work in progress please mark as draft.

**Comment by jasonnutter** [comment]
Status on this PR? And curious, how often do we get applications accidently providing an empty string for authority?

File: lib/msal-node/test/config/ClientConfiguration.spec.ts [hunk 3]
```diff
@@ -69,6 +69,28 @@ describe('ClientConfiguration tests', () => {
...
```
Comment (Robbie-Microsoft) [inline lib/msal-node/test/config/ClientConfiguration.spec.ts:76]: Should we be creating a constant for " "?

**Comment by peterzenz** [comment]
> Status on this PR? And curious, how often do we get applications accidently providing an empty string for authority?

@mpminayo updates here?

**Review by bktowett — APPROVED** [review]

**Comment by jasonnutter** [comment]
Thinking about this, I'm not sure we really need this logic. I'd rather not write code against configuration values that would intuitively not be expected to work.

**Comment by mpminayo** [comment]
From @jasonnutter's comments, I am closing this for now.

**Pull request closed without merging**